src/OpenSpaceToolkit/Astrodynamics/Trajectory/Sequence.cpp

```diff
@@ -11,31 +13,166 @@ namespace astro
+        states.add(segmentStates);
     }
 
     return states;
 }
 
+Mass Sequence::Solution::getInitialMass() const
+{
+    return segmentSolutions.accessFirst().getInitialMass();
+}
+
+Mass Sequence::Solution::getFinalMass() const
+{
+    return segmentSolutions.accessLast().getFinalMass();
+}
+
+Duration Sequence::Solution::getPropagationDuration() const
+{
+    return accessEndInstant() - accessStartInstant();
+}
+
+Mass Sequence::Solution::computeDeltaMass() const
+{
+    return Mass::Kilograms(getInitialMass().inKilograms() - getFinalMass().inKilograms());
+}
+
+Real Sequence::Solution::computeDeltaV(const Real& aSpecificImpulse) const
+{
+    Real deltaV = 0.0;
+
+    for (const auto& segmentSolution : segmentSolutions)
+    {
+        deltaV += segmentSolution.computeDeltaV(aSpecificImpulse);
+    }
+
+    return deltaV;
+}
+
+void Sequence::Solution::print(std::ostream& anOutputStream, bool displayDecorator) const
+{
+    if (displayDecorator)
+    {
+        ostk::core::utils::Print::Header(anOutputStream, "Sequence Solution");
+    }
+
+    ostk::core::utils::Print::Line(anOutputStream)
+        << "Execution is complete: " << (executionIsComplete ? "True" : "False");
+
+    ostk::core::utils::Print::Separator(anOutputStream, "Segment Solutions");
+
+    bool hasManeuver = false;
+    for (const auto& segmentSolution : segmentSolutions)
+    {
+        segmentSolution.print(anOutputStream, false);
+
+        if (segmentSolution.segmentType == Segment::Type::Maneuver)
+        {
+            hasManeuver = true;
+        }
+
+        ostk::core::utils::Print::Separator(anOutputStream);
+    }
+
+    ostk::core::utils::Print::Separator(anOutputStream, "Summary");
+
+    ostk::core::utils::Print::Line(anOutputStream) << "Start Instant: " << accessStartInstant().toString();
+    ostk::core::utils::Print::Line(anOutputStream) << "End Instant: " << accessEndInstant().toString();
+    ostk::core::utils::Print::Line(anOutputStream) << "Propagation Duration: " << getPropagationDuration().toString();
+
+    if (hasManeuver)
+    {
+        ostk::core::utils::Print::Line(anOutputStream) << "Initial Mass: " << getInitialMass().toString();
+        ostk::core::utils::Print::Line(anOutputStream) << "Final Mass: " << getFinalMass().toString();
+        ostk::core::utils::Print::Line(anOutputStream) << "Total Mass consumed: " << computeDeltaMass().toString();
+    }
+
+    if (displayDecorator)
+    {
+        ostk::core::utils::Print::Footer(anOutputStream);
+    }
+}
+
+std::ostream& operator<<(std::ostream& anOutputStream, const Sequence::Solution& aSolution)
+{
+    aSolution.print(anOutputStream);
+
+    return anOutputStream;
+}
+
 Sequence::Sequence(
     const Array<Segment>& aSegmentArray,
     const Size& aRepetitionCount,
     const NumericalSolver& aNumericalSolver,
     const Array<Shared<Dynamics>>& aDynamicsArray,
-    const Duration& maximumPropagationDuration
+    const Duration& maximumPropagationDuration,
+    const Size& verbosity
```

Comment:
This is coming from me just not being familiar with how we manage verbosity levels in OSTk.
Is the standard pattern to leave this up to the constructor?

I see a `NumericalSolver::LogType` being used elsewhere.

Is verbosity not something that can be configured for the "library as a whole".

My concern that we have an integer being translated to a log type only here, which might get a bit "cryptic" from a user perspective as you know need to give an integer. Which number should I pick?

Anyways, as I said, just maybe me not knowing how we deal with verbosity levels : )